Marlin/Marlin_main.cpp

```diff
@@ -9814,22 +9813,21 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
 
     #else // HOTENDS <= 1
 
+      UNUSED(fr_mm_s);
+      UNUSED(no_move);
+
       if (tmp_extruder >= EXTRUDERS)
         return invalid_extruder_error(tmp_extruder);
 
       // Set the new active extruder
       active_extruder = tmp_extruder;
 
-      UNUSED(fr_mm_s);
-      UNUSED(no_move);
-
-      #if ENABLED(SWITCHING_EXTRUDER)
-        stepper.synchronize();
-        move_extruder_servo(active_extruder);
-      #endif
-
     #endif // HOTENDS <= 1
 
+    #if ENABLED(SWITCHING_EXTRUDER) && !SWITCH_TEST
+      move_extruder_servo(active_extruder);
+    #endif
```

Comment:
This servo can move last, just before the next printing move.
When SWITCH_TEST is not defined it evaluates like 0.